File: app/src/main/java/com/example/josycom/flowoverstack/viewmodel/SearchViewModel.java
package com.example.josycom.flowoverstack.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import com.example.josycom.flowoverstack.model.Question;
import com.example.josycom.flowoverstack.repository.SearchRepository;

import java.util.List;

public class SearchViewModel extends ViewModel {

    private SearchRepository mSearchRepository;
    private MutableLiveData<String> mSearchLiveData = new MutableLiveData<>();
    private LiveData<List<Question>> mQuestionLiveData = Transformations.switchMap(mSearchLiveData, (query) -> mSearchRepository.getQuestions(query));

    public SearchViewModel() {
        mSearchRepository = new SearchRepository();
    }

    public LiveData<List<Question>> getQuestionLiveData() {
        return mQuestionLiveData;
    }

    public void setQuery(String query) {
        mSearchLiveData.setValue(query);
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/viewmodel/SearchViewModel.kt
package com.josycom.mayorjay.flowoverstack.ui.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import com.josycom.mayorjay.flowoverstack.data.SearchRepository
import javax.inject.Inject

class SearchViewModel @Inject constructor(private val page: Int, private val pageSize: Int, private val mSearchRepository: SearchRepository) : ViewModel() {

    private val mSearchLiveData = MutableLiveData<String>()
    val responseLiveData = Transformations.switchMap(mSearchLiveData) { query: String? -> mSearchRepository.getResponse(query, page, pageSize) }

    fun setQuery(query: String) {
        mSearchLiveData.value = query
    }
}